rocks/RocksDatabase.java

```diff
@@ -278,6 +286,26 @@ public ReasonerCache reasoner() {
         public SchemaGraph schemaGraph() {
             return schemaGraph;
         }
+
+        public void borrow() {
```

Comment:
We need to put the cache close in `RocksDatabase.close()` into this new structure.